static/js/components/dashboard/CourseAction_test.js

```diff
@@ -63,11 +63,13 @@ describe('CourseAction', () => {
     const wrapper = shallow(<CourseAction course={course} now={now} checkout={checkoutStub}/>);
     let buttonContainer = wrapper.find(".course-action-action");
     let description = wrapper.find(".course-action-description");
-    let buttonProps = buttonContainer.find("button").props();
+    let buttonProps = buttonContainer.find(".dashboard-button").props();
+    let spanText = buttonContainer.find(".sr-only").text();
+    let buttonText = buttonContainer.find(".dashboard-button").children().text();
     let firstRun = course.runs[0];
 
     assert(!buttonProps.disabled);
-    assert.equal(buttonContainer.text(), `Upgrade in ${firstRun.title}`);
+    assert.equal(`${buttonText}${spanText}`, `Upgrade in ${firstRun.title}`);
```

Comment:
a few changes for these unit tests:

1) change `assertCheckoutButton` to accept a wrapped button object
```javascript
  let assertCheckoutButton = (button, courseId) => {
    button.simulate('click');
    assert.equal(checkoutStub.callCount, 1);
    assert.deepEqual(checkoutStub.args[0], [courseId]);
  };
```

2) create a new test case for the course title so we don't need to repeat it 6 times (the text doesn't change for any of these cases)
```javascript
it('shows the course title near the enroll button', () => {
    let course = findCourse(course => course.status === STATUS_ENROLLED_NOT_VERIFIED);
    const wrapper = shallow(<CourseAction course={course} now={now} checkout={checkoutStub}/>);
    let buttonContainer = wrapper.find(".course-action-action");
    let firstRun = course.runs[0];
    assert.include(buttonContainer.text(), firstRun.title);
  });
```

3) change the pattern of these unit tests to match this (i think it's cleaner, clearer, and slightly less repetitive)
```javascript
  it('shows an upgrade button if user is not verified but is enrolled', () => {
    let course = findCourse(course => course.status === STATUS_ENROLLED_NOT_VERIFIED);
    const wrapper = shallow(<CourseAction course={course} now={now} checkout={checkoutStub}/>);
    let buttonContainer = wrapper.find(".course-action-action");
    let description = wrapper.find(".course-action-description");
    let button = buttonContainer.find(".dashboard-button");
    let firstRun = course.runs[0];

    assert(!button.is(':disabled'));
    assert.equal(button.children().text(), 'Upgrade');
    assert.equal(description.text(), "");
    assertCheckoutButton(button, firstRun.course_id);
  });
```